### ISSUE TITLE
Slicing a view

### ISSUE THREAD

**Issue by bernhardmgruber** [alpaka-group/llama#332]
LLAMA's views give full access to the data space spaned by the record and array dimensions. But sometimes we want to restrict this space to a subspace/slice where certain coordinates in the array/record dimension are already bound or restricted. LLAMA should have a corresponding feature.

Specifically, LLAMA should allow to create a new view from an existing view, while:
1. binding the value of the array domain to a specific value. The returned view has 1 array dimension less
2. restricting the range of array dimension indices in 1 array dimension. An access outside this range is a runtime error. Whether the indices on the restricted range are shifted or not still needs to be discussed.
3. binding the value of the record dimension. Each access using `operator()(ArrayDims)` on the returned view is immediately terminal.
4. restricting the record dimension to a subset. Accessing a restricted record dimension field on the returned view results in a compile time error.

LLAMA's `VirtualView` already provides a feature for shifted access. We could extend `VirtualView` with these new features or replace it.

Restricting the array dimensions also raises the questinons whether the array dimensions' runtime sizes should be queryable at the view. Currently, this is often done via `view.mapping.arrayDims()`, but this always returns the real sizes, violates encapsulation and gives us no means to change it, e.g. for restricted views. LLAMA's `View` therefore probably needs an `arrayDims()` observer, which by default returns `mapping.arrayDims()`.

**Comment by bussmann** [comment]
Of course, this goes in the same direction as ranges for arrays, but you know this already ;-)

**Comment by bernhardmgruber** [comment]
Here is the most recent specification of `submdspan`, which has a very similar feature set: https://wg21.link/p2630.
